src/services/review/ReviewRequestService.ts

```diff
@@ -134,96 +114,108 @@ export default class ReviewRequestService {
           )
         )
-      )
+        return ResultAsync.combine(editMetadata)
+      })
       .map((changedItems) =>
         changedItems.filter(
           (changedItem): changedItem is WithEditMeta<DisplayedEditedItemDto> =>
             changedItem.type !== "placeholder"
         )
       )
 
-  private compareDiffWithMappings = (
+  createEditedItemDtoWithEditMeta = (
+    filename: string,
     sessionData: UserWithSiteSessionData,
-    stagingLink: StagingPermalink,
-    files: RawFileChangeInfo[],
-    mappings: ShaMappings
-  ): ResultAsync<WithEditMeta<EditedItemDto>, BaseEditedItemDto>[] => {
-    // Step 1: Get the site name
+    stagingLink: StagingPermalink
+  ): ResultAsync<WithEditMeta<EditedItemDto>, never> => {
     const { siteName } = sessionData
+    const editMeta = this.extractEditMeta(siteName, filename)
+    const editedItemInfo = this.extractEditedItemInfo(
+      filename,
+      sessionData,
+      stagingLink
+    )
 
-    return files.map(({ filename, contents_url: contents }) => {
-      const extractedPathResult = extractPathInfo(filename)
-      if (extractedPathResult.isErr()) {
-        return errAsync({
-          name: "",
-          path: [],
-          type: "file",
-        })
-      }
-
-      const pathInfo = extractedPathResult.value
+    return ResultAsync.combine([editMeta, editedItemInfo]).map(
+      ([editMetadata, item]) => ({
+        ...item,
+        ...editMetadata,
+      })
+    )
+  }
 
-      return this.extractConfigInfo(pathInfo)
-        .orElse(() => this.extractPlaceholderInfo(pathInfo))
-        .orElse(() => this.extractMediaInfo(pathInfo))
-        .asyncMap<EditedItemDto>(async (item) => item)
-        .orElse(() =>
-          this.extractPageInfo(pathInfo, sessionData, stagingLink, siteName)
-        )
-        .map<WithEditMeta<EditedItemDto>>((item) => {
-          const items = contents.split("?ref=")
-          // NOTE: We have to compute sha this way rather than
-          // taking the file sha.
-          // This is because the sha present on the file is
-          // a checksum of the files contents.
-          // And the actual commit sha is given by the ref param
-          const hash = items[items.length - 1]
-          return {
-            ...item,
-            lastEditedBy: mappings[hash]?.author || "Unknown user",
-            lastEditedTime: mappings[hash]?.unixTime || 0,
-          }
+  extractEditMeta = (siteName: string, filename: string) =>
+    this.apiService
+      .getLatestLocalCommitOfPath(siteName, filename)
+      .andThen((latestLog) => {
+        const { userId } = fromGithubCommitMessage(latestLog.message)
+        return ResultAsync.fromPromise(
+          this.users.findByPk(userId),
+          () => new DatabaseError()
+        ).map((author) => ({
+          lastEditedBy: author?.email || latestLog.author_email,
+          lastEditedTime: new Date(latestLog.date).getTime(),
+        }))
+      })
+      .orElse(() =>
+        ok({
```

Comment:
is this expected ?

Reply:
should we log a warn msg here?

Reply:
yea this actually follows current behaviour, but yes can definitely add logs here

Reply:
[cd07757](https://github.com/isomerpages/isomercms-backend/pull/1172/commits/cd077572994c84265dfb274d2504dedafe546f65)